### ISSUE TITLE
Help with hard debrick :)

### ISSUE THREAD

**Issue by sec** [OnionIoT/omega2-bootloader#8]
Hi,
I've been playing around with mine Omega2p+ and had erased u-boot (performed cp.uboot without loading anything to write) - so I assume it zero'ed first 4 bytes of u-boot, looking at the output.
Now device don't boot, serial output is blank, on power connect, the led on expansion dock blink for a while, then it's off.

Is there any way to debrick, maybe boot from microsd card or reflash u-boot without desoldering shield and reprogramming flash directly on the board?

**Comment by greenbreakfast** [comment]
Hi @sec 
Uff this is a tough situation to be in. With no bootloader, there's no way to boot from internal flash or external storage.

To recover your device, you don't have to desolder the shield. Instead you can use the SPI pins to reprogram the flash. Just note you'll need to copy the entire flash from another, working Omega2+ in order to have an image to flash.

Check out this post on the Onion community forum for details about writing directly to the flash: https://community.onion.io/topic/5011/spi-flash-image-generation/4

Good luck!

**Comment by sec** [comment]
Thanks for the tip. I've already ordered second unit, just in case.

But to make this one alive. Looking at [Omega2+ pinout](https://docs.onion.io/omega2-docs/omega2p.html), I see 4 SPI pins to connect, two questions:
1) which one is VDD_FLASH - is it VIN 3.3v ?
2) which GND should I use?

The link you gave is about 2S+ and my unit is Omega2 Plus :)

**Comment by greenbreakfast** [comment]
> 1\. which one is VDD_FLASH - is it VIN 3.3v ?

VDD_FLASH is only exposed on the Omega2S/2S+, it provides power only to the flash. Hmm, you may not be able to write directly to the flash on the through-hole Omega2/Omega2+. If you provide power to the 3.3V pin then the CPU will occupy the SPI lines.

I'll double check on this and get back to you

> 2\. which GND should I use?

Either/both. Theyre connected!

**Comment by sec** [comment]
Yes, after more reading/learning I see that there is no CS0 exposed on Omega2+ and there's no VDD_FLASH exposed also... So it's by design unable to reflash uboot without opening RF shielf and getting to the pins of the flash directly :(

**Comment by greenbreakfast** [comment]
@sec Yes you're correct, I had forgotten about CS0, my mistake. 
Unfortunately it is not possible to reprogram the flash on an Omega2/Omega2+ without removing the RF shielding.
